### ISSUE TITLE
Adapter not "fired" inside a click event

### ISSUE THREAD

**Issue by Lupici** [amcharts/amcharts4#3760]
I all, I try to do a simple thing (I think...). Change the color of the bars dinamically after a click (on a ColumnSeries). The bar clicked remains with the same color, the others became gray. So I think to use Adapter.

This is my code:

this.barseries = this.chart.series.push(new am4charts.ColumnSeries());
....
....
....

//Handle Click on bar
this.barseries.columns.template.events.on("hit", function(ev) {
    var refDate=ev.target.dataItem.dataContext.date;
    console.log(refDate);
    clickedBar(refDate);
}, this);

function clickedBar(refDate) {
    //self.chart.invalidateData();
    self.barseries.columns.template.adapter.add("fill", function(fill, target) {
        console.log("HERE");
        if (target.dataItem && (target.dataItem.dataContext.date == refDate)) {
              return fill;
        }
        else {
              return am4core.color("#555555");
        }
    });	
}

When I click on a bar, it enters on "clickedBar" function, but nothing else happend. It never enters inside the adapter.add function. 
If I un-comment the "self.chart.invalidateData();" line, forcing the graph to redraw (with bad effets due to animation, because the data are the same..), IT ENTERS on adapter.add function..... But I need to enter without the "invalidate".....

Hope you can help me

Thanks

L.

**Comment by martynasma** [comment]
In order to trigger fill adapter you may need to access its `fill` property:

```JavaScript
ev.target.fill = ev.target.fill;
```

**Comment by Lupici** [comment]
> In order to trigger fill adapter you may need to access its `fill` property:
> 
> ```js
> ev.target.fill = ev.target.fill;
> ```

Sorry, I don't understand, you reply sound me like 1=1..  :)
Maybe you are saying me that I have to look for ev.target.fill when I click.... ok.. but I have to change the color of ALL the bars.... The one I clicked remains the same, the others change color. If I use ev.target, I'm acting only on the one I clicked, I think

**Comment by martynasma** [comment]
Nope, just try adding this line :)

Normally, changing something does not invoke adapter that would change actual fill.

However accessing `ev.target.fill` DOES invoke adapter, which then returns proper color as per adapter code, which is then assigned to `fill` property.

Hope that makes sense.

**Comment by Lupici** [comment]
> Nope, just try adding this line :)
> 
> Normally, changing something does not invoke adapter that would change actual fill.
> 
> However accessing `ev.target.fill` DOES invoke adapter, which then returns proper color as per adapter code, which is then assigned to `fill` property.
> 
> Hope that makes sense.

Yes now it makes a bit more sense, but I try and it doesn't work

		this.barseries.columns.template.events.on("hit", function(ev) {
			var refDate=ev.target.dataItem.dataContext.date;
			//ev.target.fill = ev.target.fill;
			console.log(refDate);
			clickedBar(refDate);
			ev.target.fill = ev.target.fill;
		}, this);

I try to put the line before the call to clickedBar, and after the call. No changes in behaviour.... it never enters che adapter....

**Comment by martynasma** [comment]
In that case you must check the code for your adapter, why it's producing wrong `fill` value.

**Comment by Lupici** [comment]
> In that case you must check the code for your adapter, why it's producing wrong `fill` value.

The code I use is the one written in this post, no changes...... :(

**Comment by martynasma** [comment]
In that case, can you post your whole chart in CodePen or jsFiddle so that we could live-test it?

**Comment by Lupici** [comment]
> In that case, can you post your whole chart in CodePen or jsFiddle so that we could live-test it?

Ok, I will try!

**Comment by Lupici** [comment]
Here is a Codepen code.... 

https://codepen.io/Lupix/pen/powwxPa

**Comment by martynasma** [comment]
OK, this does not seem right.

You are adding a new adapter every time bar is clicked.

I don't think you want that.

You want to add adapter only once, then trigger it when click happens:

https://codepen.io/team/amcharts/pen/d5c6393325ed5655f0ce1c9d6816eb1b?editors=0010
